pkg/pcap/cache.go

```diff
@@ -0,0 +1,14 @@
+package pcap
+
+import (
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+//
+// Multiple Pcap Implementation Caches
+//
+
+type pcapCache interface {
```

Comment:
Now for something more substantive.
The interface here is redundant and can be replaced with one `struct` using a generic type argument whose constraint is the `pcap` interface.
Unless I missed something all the caches functionally do exactly the same thing except for type checking the pcap object stored. Using a generic implementation of Cache you can use the `T` type to do the type checking according to the type provided when constructing the cache. This way we reduce the 3 separate implementations to one.

Reply:
There is a TODO for that in the code. I'll do it right after, in a new PR, if you don't mind.

Reply:
FWIW I think doing this for the cache right now would be worth it, it should be pretty easy and quick, for the pcaps maybe less so.

Reply:
I'll give it a try.

Reply:
Actually, you spotted a bug. The caches should be indexed differently:

- process cache entry using host_tid
- container cache entry using container_id
- command cache entry using process_name (or comm)

so... they wouldn't be "exactly" the same but still ok for generics, although I don't like the complexity added in a first attempt. I think I'll stick with common functions and minor implementations (easier to understand and maintain).